src/announcement/announcement.ts

```diff
@@ -0,0 +1,85 @@
+//eslint-disable-next-line
+require("dotenv").config();
+import Web3 from "web3";
+import { Contract } from "web3-eth-contract";
+import { AbiItem } from "web3-utils";
+
+import { KeccakHash } from "../types/hash";
+import { hashPrefix } from "../utilities/hashPrefix";
+
+const BATCH_CONTRACT_ADDRESS = String(process.env.BATCH_CONTRACT_ADDRESS);
+const BATCH_CONTRACT_ABI: AbiItem[] = [
+  {
+    anonymous: false,
+    inputs: [
+      {
+        indexed: false,
+        internalType: "bytes32",
+        name: "hash",
+        type: "bytes32",
+      },
+      {
+        indexed: false,
+        internalType: "string",
+        name: "dsnpUri",
+        type: "string",
+      },
+    ],
+    name: "DSNPBatch",
+    type: "event",
+  },
+  {
+    inputs: [
+      {
+        internalType: "bytes32",
+        name: "hash",
+        type: "bytes32",
+      },
+      {
+        internalType: "string",
+        name: "dsnpUri",
+        type: "string",
+      },
+    ],
+    name: "batch",
+    outputs: [],
+    stateMutability: "nonpayable",
+    type: "function",
+  },
+];
+
+const contract: Contract | null = null;
+
+const getContract = (web3Instance: Web3): Contract => {
+  if (contract) return contract;
+  return new web3Instance.eth.Contract(BATCH_CONTRACT_ABI, BATCH_CONTRACT_ADDRESS);
+};
+
+const getGasLimit = async (contract: Contract, uri: string, hash: KeccakHash, fromAddress: string): Promise<number> => {
+  const gasEstimate = await contract.methods.batch(hashPrefix(hash), uri).estimateGas({
+    from: fromAddress,
+    gas: 27000,
+  });
+
+  return gasEstimate + 2000;
```

Comment:
1. 2000 feels like a large buffer. Does it really need to be that large?
2. Can we put the 2000 into a constant?